pkg/controller/deployment/deployment_controller.go

```diff
@@ -799,28 +811,97 @@ func (dc *DeploymentController) reconcileOldRCs(allRCs []*api.ReplicationControl
 	}
 	// Check if we can scale down.
 	minAvailable := deployment.Spec.Replicas - maxUnavailable
+	maxScaledDown := allPodsCount - minAvailable
```

Comment:
Same example, but pushing a new pod template instead of rolling back (aka "roll over"):

The new RC created must start with 0 replicas because allPodsCount is already at 13. However, newRCPodsUnavailable would also be 0, so the 2 old RCs could be scaled down by 5 (13 - 8 - 0), which would then allow the new RC to be scaled up by 5.

Great. That's the behavior I'd want.